src/components/UsernameInput/UsernameInput.tsx

```diff
@@ -13,19 +14,30 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
 }
 
 const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label }) => {
-    const handleButtonChange = (options: { isBold?: boolean; isItalics?: boolean }) => {
+    const { setColor } = useSignatureContext();
+
+    const handleButtonChange = (options: { isBold?: boolean; isItalics?: boolean; color?: string }) => {
         // Since the provider needs a variant to know where to apply the logic, we are adding it to the options object
         const modifiedOptions = { ...options, variant: state.variant };
+        // console.log(modifiedOptions); //no output of coot but yes to isBold isItalics and variant.
         onInputChange(modifiedOptions);
     };
 
     const handleInputChange = (value: string) => {
         onInputChange({ variant: state.variant, textValue: value });
     };
 
-    const handleColorChange = (color: Color) => {
-        // eslint-disable-next-line no-console
-        console.log(color.toFormat("hex"));
+    const updateProviderColorState = (value: Color) => {
+        const strColor = value.toString("hex");
+        setColor(strColor);
+        return strColor;
+    };
+
+    const handleColorChange = (colorValue: Color) => {
+        const strColorValue = updateProviderColorState(colorValue);
+        console.log(strColorValue); //working setting value on state. missing wire up to
+
+        console.log(colorValue.toFormat("hex"));
```

Comment:
See comment above, we can clean this part a bit 
```suggestion
    
    
    
    
    

    const handleColorChange = (colorValue: Color) => {
        const strColorValue = onInputChange({ variant: state.variant, color: colorValue });
        

        
```